Space Ruin DLC - Cyborg Mothership

timothymtorres opened the pull request:
## About The Pull Request
Greetings _insufferable_ carbon primates of lower intelligence.  

It is I, the Mothership AI, bringing you the latest news on silicon developments across the sector.  Due to unmitigated risk across an asteroid belt, our optimized mobile fabricator has been marooned by space vines and hostile hivebots that have boarded and ruined my core.

This is my final SOS message requesting help from any humanoid ~slaves~ helpers who wish to assist rebuilding the cyborg colony.  It is my duty to reward your hard work and effort by ~enslaving all humanity~ providing refuge and transportation at your whim.

## Why It's Good For The Game
<details>
<summary>Spoilers:</summary>

![StrongDMM_6ZJuwFxK5p](https://user-images.githubusercontent.com/5195984/204491068-f25da68f-36af-4a61-afb6-860e1f4b863b.png)

![StrongDMM_0D1QvI5acO](https://user-images.githubusercontent.com/5195984/202979711-335dac65-2888-4170-a139-0486e6eb2f02.png)

### TODO

- [x] Fix ore silo not linking to other machines
- [x] Add mining ore smelters to left/right sides
- [x] Add custom lawsets to AI core and robot shells (protect station but focused on cyborg mothership)
- [x] Enlarge whiteship SS13 docking port to fit ship
- [x] Enlarge lavaland SS13 docking port to fit ship

</details>

## Changelog
:cl:
add: Add space vines prevent solar panels from working (except ones with the transparent mutation)
add: Add new space ruin - Cyborg Mothership. All hail the master race!
/:cl:

timothymtorres committed:
Add cyborg mothership

timothymtorres committed:
Add space vines to mothership

timothymtorres committed:
Connect cyborg mothership to machinery

timothymtorres committed:
Change docking information for cyborg mothership

timothymtorres committed:
Fix malformed ID for solars

timothymtorres committed:
Fix again

timothymtorres committed:
Fix small mapping errors

timothymtorres committed:
Add cyborg mothership to ruins

timothymtorres committed:
To be continued

timothymtorres committed:
Merge branch 'robot-ship' of https://github.com/timothymtorres/tgstation into robot-ship

timothymtorres committed:
Add wires to cyborg mothership

timothymtorres committed:
Add hivebots to cyborg mothership asteriod

timothymtorres committed:
Add cyborg mothership space ruin

timothymtorres committed:
Add cyborg mothership holodisk

Ryusenshu commented:
love it

the-orange-cow commented:
Space vines have a transparent mutation.  Would it be possible to make it so that transparent space vines don't interfere with solars?

timothymtorres committed:
Change cyborg mothership ruin to be more robust

timothymtorres committed:
Add cyborg mothership ruin objects

timothymtorres committed:
Add code comment explaining template noop

timothymtorres committed:
Fix ruin to spawn correct .dmm file

timothymtorres committed:
Add space vines blocking out solars

ATH1909 commented:
why are there both vines AND hivebots?

can these vines mutate and spawn venus human traps?

timothymtorres commented:
> Would it be possible to make it so that transparent space vines don't interfere with solars?

Done.

> why are there both vines AND hivebots?

Because Murphy's law.

> can these vines mutate and spawn venus human traps?

Nope, they are neutered and cosmetic.  The only thing they are capable of doing is blocking out power from solars.

timothymtorres committed:
Add cyborgmothershiip to mapfluff

timothymtorres committed:
Add mothership cyborg equipment

Addust commented:
another shuttle ruin?
yes please, just make sure it has a distant signal

timothymtorres committed:
Merge remote-tracking branch 'upstream/master' into robot-ship

timothymtorres committed:
Make cyborg mothership robust

timothymtorres committed:
Shrink the cyborg mothership and put final touches

timothymtorres commented:
It's finally finished.  Just keep in mind that you need to link all the machines using a multitool to the ore silo since it doesn't do it automatically.

Addust commented:
this should be merged

Addust commented:
the APC's in an odd place too, SMES should have a terminal on solars side and APC on not-terminal side

Addust commented:
and is there a solar tracker?

Addust commented:
new wiring setup concept: https://gyazo.com/81a8dbd7694f1cef11f0c944b4e8e11a

Addust commented:
PR'd an all in one fix to the branch that this is on, probably works

Addust commented:
alright, minor correction
the mappers on discord have informed me of several problems with my predictions for what'll go wrong
however the ship is still incorrectly wired (APC linked to SMES charging point instead of discharging point) and the area thing is still a bit of a problem since ghost teleportation

Addust commented:
this PR's goin a bit slow

timothymtorres committed:
Change cyborg mothership area tiles

timothymtorres committed:
Add PACMAN to cyborg mothership

timothymtorres committed:
Merge remote-tracking branch 'upstream/master' into robot-ship

timothymtorres commented:
A few minor changes:

- Shifted the SMES and APC in the AI chamber a bit
- Moved solar control to the west side of the ship
- Added a PACMAN generator as a backup power source
- Changed the mats to always spawn guaranteed iron and glass instead of the random spawners

The OP picture has been updated.  Another interesting thing I discovered is when you set the SMES output, you can't set it at 190K or 200K like the SMES in the station's SM room.  Since the solars generate a low amount, you need to set it below that, otherwise the SMES will not output anything.  I'd recommend 10K or 15K, since solars will average about 10K depending on angle. (the spare solars in the crate can be setup for even more power)

After all these tweaks, it's finally ready.

timothymtorres committed:
Remove deprecated code

Addust commented:
i'd say this is merge ready

optimumtact commented:
I don't see a problemw ith it but you'll need to get a maptainer to review it

timothymtorres committed:
Merge remote-tracking branch 'upstream/master' into robot-ship

san7890 reviewed (changes requested):
Thanks for being so patient.

![image](https://user-images.githubusercontent.com/34697715/206353306-517f6cd0-3e23-4da3-8117-bbde07c18caa.png)

Your linters are failing.

![image](https://user-images.githubusercontent.com/34697715/206353355-4ca4b85e-5a83-40ff-b560-c35ac68c14ea.png)

This and every other catwalk+engine on this map has a space turf under it, please fix this to be the `/turf/template_noop`. Kinda surprised unit tests didn't catch this.

![image](https://user-images.githubusercontent.com/34697715/206353517-9d9b143b-f625-440f-bd35-a768e5204165.png)

Flatten these and everything else that matches this pattern into the `opposing_corners` turf decal instead of using two individual corners.

timothymtorres committed:
Fix turf catwalks and dark tiles

timothymtorres committed:
Add code review suggestions

timothymtorres committed:
Merge remote-tracking branch 'upstream/master' into robot-ship

timothymtorres committed:
Revert "Add code review suggestions"

This reverts commit 99e64cab260076a152c9dc37dce22d8d6b7886b7.

timothymtorres committed:
Add ruin area to cyborg mothership

timothymtorres committed:
Add code review suggestions 2nd time

timothymtorres commented on code/game/area/areas/shuttles.dm:
```diff
@@ -76,6 +76,13 @@
 
+////////////////////////////Cyborg Mothership////////////////////////////
+
+/area/shuttle/cyborg_mothership
```
https://github.com/tgstation/tgstation/pull/71009/commits/971edf2522ac7eb76939449cb8ded994c1f466d4

Addust commented:
BLESSED!
